modules/data.atmosphere/R/download.ECMWF.R

```diff
@@ -0,0 +1,129 @@
+#' Download ECMWF Open Data 15 day forecast data products
+#'
+#' @author Swarnalee Mazumder
+#'
+
+library(reticulate)
+# py_install("ecmwf-opendata",pip = TRUE)
+
+download.ECMWF <- function(outfolder, 
+                           fchour, 
+                           stream, 
+                           type = "pf", 
+                           parameters = "all", 
+                           overwrite = FALSE, 
+                           reticulate_python = NULL,...)
+  
+{
+  if (!file.exists(outfolder)) {
+    dir.create(outfolder, showWarnings = FALSE, recursive = TRUE)
+  }
+  
+  
+  tryCatch({
+    ecmwfod <- reticulate::import("ecmwf.opendata")
+  }, error = function(e) {
+    PEcAn.logger::logger.severe(
+      "Failed to load `ecmwfod` Python library. ",
+      "Please make sure it is installed to a location accessible to `reticulate`.",
+      "You should be able to install it with the following command: ",
+      "`pip install --user ecmwf-opendata`.",
+      "The following error was thrown by `reticulate::import(\"ecmwf.opendata\")`: ",
+      conditionMessage(e)
+    )
+  })
+  
+  hour <- fchour
+  
+  all_hours <- c(00, 12)
+  
+  if (any(!hour %in% all_hours)) {
+    bad_hours <- setdiff(hour, all_hours)
+    PEcAn.logger::logger.severe(sprintf(
+      "Invalid forecast hours  %s. 15 day forecast hours must be one of the following: %s",
+      paste0("`", bad_hours, "`", collapse = ", "),
+      paste0("`", all_hours, "`", collapse = ", ")
+    ))
+  }
+  
+  if (fchour == 00){
+    time <- 0
+  } else if (fchour == 12){
+    time <- 12
+  }
+  
+  all_parameters <- c("2t", "tp", "10u", "10v", "q", "r", "sp")
+  
+  if (tolower(parameters) == "all") {
```

Comment:
if this condition is not met there will be no `parameter_types`, then line 61 will most likely throw a `'parameter_types' not found` error, should there be an `else` to this `if`?

Reply:
Added `else` to show what can be given as an input, here `all`.

Reply:
I believe this is remains to be the case, please add an `else` to make sure `parameter_types` are assigned something in case `tolower(parameters) != "all"`

Reply:
If `tolower(parameters) != "all"` should the `parameter_types` still be `"all"`?